### ISSUE TITLE
Vanilla Recipe Book

### ISSUE THREAD

**Issue by vaporvee** [WolfyScript/CustomCrafting-Wiki#70]
**Is your feature request related to a problem? Please describe.**
I have one Plugin "Instruments" wich has custom recipes in the vanilla recipe book but this plugin not

**Describe the solution you'd like**
I'd like my recipes for my server project in one easy accessable recipe book

**Describe alternatives you've considered**
i don't want alternatives

**Additional context**
https://www.spigotmc.org/resources/instruments.87156/
<img src="https://cdn.discordapp.com/attachments/825019604207927326/900055144375386123/Screenshot_2021-10-19_181653.png" height="200">

**Comment by WolfyScript** [comment]
The reason the recipes are not shown in that book is that there are some limitations to the vanilla recipe book, which kinda make the book useless and buggy.

The biggest issue is that it doesn't support ingredients with NBT, the last time I checked. 
Instead, if you tried to complete the recipe, it put the wrong items into the grid and was very buggy. 
See https://bugs.mojang.com/browse/MC-122622, which is marked invalid, as it doesn't support NBT.

The other issue is that if you have too many recipes (or your recipes contain too many items, with NBT etc.) the players won't be able to join. See https://bugs.mojang.com/browse/MC-185901

So If I think about it, I could allow the recipes that only contain material ingredients (without NBT) into the recipe book, or add a toggle for all recipes. That would allow you to add the recipes you want to view in the book.

**Comment by vaporvee** [comment]
In the screenshot we see nbt flags working but there are other bugs right? I'll try the instruments plugin more out but maybe it could be a experimental feature in the config but i think its not your biggest priority

**Comment by WolfyScript** [comment]
Yes they work for the results, but not for ingredients. 
For me it is quite important too, which is the reason I am working on a mod to support that and integrate a recipe book directly into the GUI, but well that would require a mod.

I will see what I can do and add a config setting for recipes.

**Comment by vaporvee** [comment]
Thanks in my case i would just use vanilla ingredients anyway

**Comment by vaporvee** [comment]
But it would also need a whitelist wich items should be put in the vanilla recipe book if you would work on this

**Comment by WolfyScript** [comment]
Yes that what I actually meant. I will add a setting for each recipe config, not just one setting in the plugins' config.
Plus maybe a toggle in the GUI.

**Comment by vaporvee** [comment]
Thank you very much!
